src/main/java/uk/gov/hmcts/reform/sendletter/logging/AppInsights.java

```diff
@@ -43,14 +51,30 @@
 
     private final TelemetryClient telemetryClient;
 
-    public AppInsights(TelemetryClient telemetryClient) {
+    private final IFtpAvailabilityChecker ftpAvailabilityChecker;
+
+    public AppInsights(
+        TelemetryClient telemetryClient,
+        IFtpAvailabilityChecker ftpAvailabilityChecker
+    ) {
         this.telemetryClient = telemetryClient;
+        this.ftpAvailabilityChecker = ftpAvailabilityChecker;
     }
 
     // schedules
 
     @Around("@annotation(org.springframework.scheduling.annotation.Scheduled)")
     public void aroundSchedule(ProceedingJoinPoint joinPoint) throws Throwable {
+        Method method = ((MethodSignature) joinPoint.getSignature()).getMethod();
+        FtpConstraint ftpConstraint = AnnotationUtils.findAnnotation(method, FtpConstraint.class);
+
+        if (ftpConstraint != null) {
+            if (!ftpAvailabilityChecker.isFtpAvailable(LocalTime.now(ZoneId.of(EUROPE_LONDON)))) {
+                log.info("Not processing '{}' task due to FTP downtime window", ftpConstraint.name());
+                return;
+            }
+        }
```

Comment:
Why is checking FTP availability a part of logging component?

To sum up this PR:
Fancy? Yes.
Clearer, better? In my opinion, no.

👎 from me.